### PR TITLE
Adding a "Deselect All" button

### PR DESCRIPTION
**Pull request by CRImier**
Adding a "Deselect All" button so that you don't need to uncheck all other sessions in a multi-session environment every time you want to save three single-window session out of, say, 50 windows.

I have 88 windows open at the moment (with 1588 tabs), and I want to clean it up some by saving sessions until they're needed. Picked your extension because you seem active when it comes to development and responses to users, and decided I'd try developing the feature I currently need the most =) Tested this and it works for me, let me know if I need to make any changes.

Cheers!
Arsenijs

### CODE REVIEW

**Comment by CRImier** [comment]
Hi! Any news on merging this? I'd really like this feature to be mainlined so that the extension can be autoupdated in the future without any problems, and so that other people can use this feature, too.

**Comment by maximstewart** [comment]
@CRImier   I apologies, I never got a notification for this PR. I'm reviewing the change.

**Comment by maximstewart** [comment]
It looks like I added this type of feature already. I apologies again for the late reply. I'll be closing the request.

**Comment by CRImier** [comment]
Hey! Thank you so much for your reply! Is this feature in the latest release? I can't find it. Currently using my custom version from GitHub with my changes. Cheers!

**Comment by maximstewart** [comment]
It is in the latest release if I've read your changes correctly and what you were going for. You wanted a (de)select togglable button/checkbox to save sessions differently. I've added that type of feature titled "Select All" with the default state as checked. But, you can uncheck it to deselect an entire window from being saved. the same exists for editing a session too.

**Comment by CRImier** [comment]
I don't think we're talking about the same thing. I have 50 windows right now. If I want to save one window into a session, I need to deselect every single window but one of them every time I do "Save", and given the UI isn't actually usable enough for scrolling through windows one-by-one, it's hard. If I need to save 10 windows out of 50 and I need to deselect windows one-by-one, I need to click **445** times - with this button, I only need to click 10 times.

Here's how the button that I added looks, to clarify:

image

**Comment by CRImier** [comment]
Basically, the "Select all" checkbox only works on a single window. I have a lot of windows and the "Select all" checkbox is not enough because, while it's a step in the right direction, I still need to click it too many times.

**Comment by maximstewart** [comment]
Ahhh, OK, I understand. Would you please edit this to work with the Edit page as well? I'll merge the request thereafter. I think you'll just need to add the same button to the Edit modal.

File: src/scripts/events.js [hunk 1]
```diff
@@ -67,6 +67,10 @@ document.addEventListener("click", (e) => {
             }
         }
 
+        else if (action.includes("deselectAll")) {
+            var container = document.getElementById("editSelectionContainer");
```
Comment (maximstewart) [inline src/scripts/events.js:71]: Please change from "var" to "let"

File: src/scripts/utils.js [hunk 2]
```diff
@@ -193,6 +193,15 @@ const getSelectionData = (container = null, keys = null, keysLength = 0) => {
     return sessionData;
 }
 
+const deselectAll = (container = null) => {
+    var selAlls = container.querySelectorAll('.selAll');
```
Comment (maximstewart) [inline src/scripts/utils.js:197]: Same as other request: "var"  --> "let"

**Comment by CRImier** [comment]
Added, tested, seems to work. Let me know if there are any other problems. Cheers!

**Comment by maximstewart** [comment]
Plugin has been updated:

    https://addons.mozilla.org/en-US/firefox/addon/easy-session-manager/

**Comment by CRImier** [comment]
Thank you!

**Pull request merged**